test/modules/TestResultCoordinator/Reports/DeploymentTestReportGenerator.cs

```diff
@@ -0,0 +1,156 @@
+        public async Task<ITestResultReport> CreateReportAsync()
+        {
+            Logger.LogInformation($"Start to generate report by {nameof(DeploymentTestReportGenerator)} for Sources [{this.ExpectedSource}] and [{this.ActualSource}]");
+
+            TestOperationResult lastActualDeploymentTestResult = null;
+            ulong totalExpectedDeployments = 0;
+            ulong totalActualDeployments = 0;
+            ulong totalMatchedDeployments = 0;
+            var unmatchedResults = new List<TestOperationResult>();
+
+            bool hasExpectedResult = await this.ExpectedTestResults.MoveNextAsync();
+            if (hasExpectedResult)
+            {
+                totalExpectedDeployments++;
+            }
+
+            bool hasActualResult = await this.ActualTestResults.MoveNextAsync();
+            if (hasActualResult)
+            {
+                lastActualDeploymentTestResult = this.ActualTestResults.Current;
+                totalActualDeployments++;
+            }
+
+            while (hasExpectedResult && hasActualResult)
+            {
+                this.ValidateDataSource(this.ExpectedTestResults.Current, this.ExpectedSource);
+                this.ValidateDataSource(this.ActualTestResults.Current, this.ActualSource);
+
+                if (this.TestResultComparer.Matches(this.ExpectedTestResults.Current, this.ActualTestResults.Current))
+                {
+                    totalMatchedDeployments++;
+
+                    hasExpectedResult = await this.ExpectedTestResults.MoveNextAsync();
+                    if (hasExpectedResult)
+                    {
+                        totalExpectedDeployments++;
+                    }
+                }
+                else
+                {
+                    hasActualResult = await this.ActualTestResults.MoveNextAsync();
+                    if (hasActualResult)
+                    {
+                        totalActualDeployments++;
+                        lastActualDeploymentTestResult = this.ActualTestResults.Current;
+                    }
+                }
+            }
+
+            while (hasExpectedResult)
+            {
+                unmatchedResults.Add(this.ExpectedTestResults.Current);
+                hasExpectedResult = await this.ExpectedTestResults.MoveNextAsync();
+                if (hasExpectedResult)
+                {
+                    totalExpectedDeployments++;
+                }
+            }
+
+            hasActualResult = await this.ActualTestResults.MoveNextAsync();
+            if (hasActualResult)
+            {
+                totalActualDeployments++;
+                lastActualDeploymentTestResult = this.ActualTestResults.Current;
+
+                while (hasActualResult)
+                {
+                    // Log message for unexpected case.
+                    Logger.LogError($"[{nameof(DeploymentTestReportGenerator)}] Actual test result source has unexpected results.");
```

Comment:
I think both logged messages here should be logged as Info or Debug

Reply:
if changed to Info or Debug, this message will not be logged if log level priority is higher than Info or Debug, e.g. Fatal or Error.  Since this is weird to have more actual results, so I think it is better to use LogError here.